packages/ra-core/src/form/useInput.spec.tsx

```diff
@@ -18,243 +21,209 @@ describe('useInput', () => {
+                            defaultValue="A title"
+                            source="title"
+                            resource="posts"
+                            validate={required()}
+                        >
+                            {props => {
+                                inputProps = props;
+                                return <div />;
+                            }}
+                        </Input>
+                    )}
+                />
+            </CoreAdminContext>
         );
 
         expect(inputProps.id).toEqual('title');
         expect(inputProps.isRequired).toEqual(true);
-        expect(inputProps.input).toBeDefined();
-        expect(inputProps.input.name).toEqual('title');
-        expect(inputProps.input.value).toEqual('A title');
-        expect(inputProps.meta).toBeDefined();
+        expect(inputProps.field).toBeDefined();
+        expect(inputProps.field.name).toEqual('title');
+        expect(inputProps.field.value).toEqual('A title');
+        expect(inputProps.fieldState).toBeDefined();
     });
 
     it('allows to override the input id', () => {
         let inputProps;
         render(
-            <Form
-                onSubmit={jest.fn()}
-                render={() => (
-                    <Input id="my-title" source="title" resource="posts">
-                        {props => {
-                            inputProps = props;
-                            return <div />;
-                        }}
-                    </Input>
-                )}
-            />
+            <CoreAdminContext dataProvider={testDataProvider()}>
+                <Form
+                    onSubmit={jest.fn()}
+                    render={() => (
+                        <Input id="my-title" source="title" resource="posts">
+                            {props => {
+                                inputProps = props;
+                                return <div />;
+                            }}
+                        </Input>
+                    )}
+                />
+            </CoreAdminContext>
         );
 
         expect(inputProps.id).toEqual('my-title');
-        expect(inputProps.input).toBeDefined();
-        expect(inputProps.input.name).toEqual('title');
-        expect(inputProps.meta).toBeDefined();
+        expect(inputProps.field).toBeDefined();
+        expect(inputProps.field.name).toEqual('title');
+        expect(inputProps.fieldState).toBeDefined();
     });
 
     it('allows to extend the input event handlers', () => {
         const handleBlur = jest.fn();
         const handleChange = jest.fn();
-        const handleFocus = jest.fn();
-        let formApi;
-
-        const { getByLabelText } = render(
-            <Form
-                onSubmit={jest.fn()}
-                render={({ form }) => {
-                    formApi = form;
 
-                    return (
+        render(
+            <CoreAdminContext dataProvider={testDataProvider()}>
```

Comment:
Note that you don't need to pass a dataProvider if you need one, CoreAdminContext has default one since a few days